architecture_records/002-acceleration-framework.md

```diff
@@ -0,0 +1,302 @@
+# Training Enhancements Framework
```

Comment:
as mentioned in another PR, it might be better to track requirements/proposals in github issues as we've done previously in these fms-related repos.

Reply:
Ah ok, Will be happy to comply if you can send me an example of such a proposal @Ssukriti .

Reply:
I don't think we have a fixed template, just regular github issues, but pasting the text of an ADR into a github issue would be fine I think

Reply:
Understand, will see what @Ssukriti thinks. 

By rendering issues, do you mean some broken links, as well as inconsistencies between inline code (e.g., `SFTTrainer` vs `[SFTTrainer]`)?

Reply:
@nairbv there are several advantages to writing an architecture decision record:
https://www.redhat.com/architect/architecture-decision-records

1. It clarifies to new contributors why a certain decision was made, helps larger community understand that . We will be regularly onboarding new members and finding multiple issues for each design made will be harder than keeping all docs in one place. 
2. At later points if we want to revisit why a certain decision was made, its easier to read a doc. The purpose of ADR is to clarify the architecture decision and why it was made , what were tradeoffs and why we selected this. 
3. its easier for us to review high level design first and propose changes as needed, before everyone wastes time creating code  and then we disagree with design. in issues we may not be able to capture so many details


As far as rending is considered, using local refernces in yml usually works . I did not find broken links in the md file.. were there any particular broken URLs? they can be fixed though

Reply:
@Ssukriti aren't those all purposes of github issues too though? I just think it's better to manage that doc through the github issues UI vs committing it to the repo. Github is a purpose built system with features for supporting a variety of development processes. Circumventing that system makes it harder to take advantage of those features.

It's great to have a template to use when creating docs. We can even add a template for github issue submission form so that if a user selects ADR it pre-populates in the UI (via an md file in .github/ISSUE_TEMPLATE). We could also add a label in the UI to differentiate these from bugs and other issues.

We also already track most design decisions and planned features through github issues, across many repos. There are currently 20 open and 25 closed issues just on this repo. Introducing a new process (that so-far only has 2-3 work items being tracked) bifurcates work tracking/planning. I think this would make it much harder to find information, since one would need to both search the UI and grep the directory. I think it would be better to copy these couple of ADRs into github issues and remove this directory to reduce confusion, rather than try to get everyone else to move or to maintain two divergent processes.

With committed files, I'm not sure how we even do things like track which are open/closed, or link to them from relevant PRs. We'd have to introduce a lot of conventions to adopt this process, conventions the rest of the OSS community isn't familiar with.

Reply:
@nairbv we use issues to track work as you said. Example issue for creating this ADR https://github.com/foundation-model-stack/fms-hf-tuning/issues/144 , which is a work item. ADR can be of many forms - markdown, box notes, various other ways of documenting architecture decisions.

We are storing **design records** here for major architecture changes, so we can easily find them and go through design decisions and consequences . It also helps with easy onboarding. New members will not know what ADRs exist to be able to search long list of issues with hopes of finding some documentation in issue comments. 

Stpring ADRs as GitHub markdowns is not a new thing for open source community. Several examples exist and multiple projects are moving towards adopting ADRs
1. https://github.com/json-schema-org/json-schema-spec/tree/main/adr
2. there are several blogs and articles on benefits of ADR and where to store them, example: https://ctaverna.github.io/adr/ -> where they are stored is main;y wikipages or GitHub Mrakdowns

we have only small number of issues so far for thsi repo, as all of us have been using internal IBM trackers . we are trying to move everyone to using open source repo for issues. We have hundreds of issues for this project in internal tracker and no one is going to search each to find all relevant ADRs. 


Anyways, decision to support ADRs for this repo and contribution guidelines was discussed extensively and the need for it was raised by executives like Raghu. Hence, even if you disagree, please agree to disagree and accept this as a decision made by project owners and let us give it a chance. Thank you

Reply:
@Ssukriti I think there was some confusion on my part re ownership of this repo/project and longer term plans for it. I'd like to avoid having ADRs (committed to) the fms and fms-extras repos but I don't still have a reason to object here.